transport.go

```diff
@@ -68,15 +67,14 @@ func NewTransport(c net.Conn, l LogFactory, wef WrapErrorFunc) Transporter {
 		cdec:             cdec,
 		log:              log,
 		wrapError:        wef,
-		encodeCh:         make(chan []byte),
-		encodeResultCh:   make(chan error),
+		encodeCh:         make(chan encodingBundle, 1),
```

Comment:
@akalin-keybase I was hoping to get your thoughts here. The goal here is to prevent a blocked `Encode` from starving other `Encode` operations, but to still maintain ordering. What size do you think the buffer should have?